src/lib/storage/storage_manager.cpp

```diff
@@ -390,141 +537,204 @@ void StorageManager::write_chunk_to_disk(const std::shared_ptr<Chunk>& chunk,
+  for (auto segment_offset_index = size_t{0}; segment_offset_index < segment_count; ++segment_offset_index) {
+    header.segment_offset_ends.emplace_back(header_data[segment_offset_index + 1]);
   }
 
   return header;
 }
 
-std::shared_ptr<Chunk> StorageManager::map_chunk_from_disk(const uint32_t chunk_offset_end, const std::string& filename,
-                                                           const uint32_t segment_count) {
+std::shared_ptr<Chunk> StorageManager::_map_chunk_from_disk(const uint32_t chunk_offset_begin,
+                                                            const uint32_t chunk_bytes, const std::string& filename,
+                                                            const uint32_t segment_count,
+                                                            const std::vector<DataType>& column_definitions) const {
   auto segments = pmr_vector<std::shared_ptr<AbstractSegment>>{};
-
   auto fd = int32_t{};
   Assert((fd = open(filename.c_str(), O_RDONLY)) >= 0, "Opening of file failed.");
 
-  const auto file_bytes = std::filesystem::file_size(filename);
+  //Calls to mmap need to be pagesize-aligned
+  const auto pagesize = getpagesize();
+  const auto difference_to_pagesize_alignment = chunk_offset_begin % pagesize;
+  const auto page_size_aligned_offset = chunk_offset_begin - difference_to_pagesize_alignment;
 
-  // TODO: Remove unneccesary map on whole file
-  const auto* map = reinterpret_cast<uint32_t*>(mmap(NULL, file_bytes, PROT_READ, MAP_PRIVATE, fd, off_t{0}));
-  Assert((map != MAP_FAILED), "Mapping of File Failed.");
+  const auto* persisted_data =
+      reinterpret_cast<std::byte*>(mmap(NULL, chunk_bytes + difference_to_pagesize_alignment, PROT_READ, MAP_PRIVATE,
+                                        fd, off_t{page_size_aligned_offset}));
+  Assert((persisted_data != MAP_FAILED), "Mapping of File Failed.");
   close(fd);
 
-  const auto chunk_header = read_chunk_header(filename, segment_count, chunk_offset_end);
+  persisted_data = persisted_data + difference_to_pagesize_alignment;
+
+  const auto chunk_header = _read_chunk_header(persisted_data, segment_count, chunk_offset_begin);
 
   for (auto segment_index = size_t{0}; segment_index < segment_count; ++segment_index) {
-    auto segment_offset_end = _chunk_header_bytes(segment_count) + chunk_offset_end;
+    auto segment_offset_begin = _chunk_header_bytes(segment_count);
 
     if (segment_index > 0) {
-      segment_offset_end = chunk_header.segment_offset_ends[segment_index - 1] + chunk_offset_end;
+      segment_offset_begin = chunk_header.segment_offset_ends[segment_index - 1];
     }
 
-    segments.emplace_back(std::make_shared<DictionarySegment<int32_t>>(map + segment_offset_end / 4));
+    resolve_data_type(column_definitions[segment_index], [&](auto type) {
+      using ColumnDataType = typename decltype(type)::type;
+      if constexpr (std::is_same<ColumnDataType, pmr_string>::value) {
```

Comment:
Add comment that we can only persist `FixedStringDictionarySegments` and therefore know that this is one.